src/include/common/scoped_timer.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <chrono>  // NOLINT
+#include "common/macros.h"
+
+namespace terrier::common {
+
+// TODO(Matt): we could templatize this for different resolutions (milliseconds, nanoseconds, etc.) but then the caller
+// needs to include the <chrono> library to pass in the correct template arguments, which this is trying to be a wrapper
+// for. However, I'm not opposed to the idea of templatizing it for this purpose if we decide milliseconds is not always
+// suitable.
+
+/**
+ * The ScopedTimer provided an easy way to collect the elapsed time for a block of code. It stores the current time when
+ * it is instantiated, and then when it is destructed it updates the elapsed milliseconds time in the value pointed to
+ * by the constructor argument.
+ */
+class ScopedTimer {
+ public:
+  ScopedTimer() = delete;
```

Comment:
this line is not necessary. Default constructor will be implicitly deleted once you declare a different one.